grpc/interceptors_test.go

```diff
@@ -357,3 +367,22 @@ func TestNoCancelInterceptor(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestClientCancelledInterceptor(t *testing.T) {
```

Comment:
Rather than adding this whole new test function, I'd just add one more case (lines 380-386) to the existing `TestClientInterceptor`. In particular, lines 377 and 378 are an exact duplicate of its "good case", so there's no need to repeat the setup and happy-path testing.